workerapi/server/setup.go

```diff
@@ -0,0 +1,90 @@
+package server
+
+import (
+	"log"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/scootdev/scoot/common/endpoints"
+	"github.com/scootdev/scoot/config/jsonconfig"
+	"github.com/scootdev/scoot/ice"
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/runner/execer"
+	"github.com/scootdev/scoot/runner/execer/execers"
+	osexec "github.com/scootdev/scoot/runner/execer/os"
+	localrunner "github.com/scootdev/scoot/runner/local"
+	"github.com/scootdev/scoot/snapshot/snapshots"
+)
+
+type servers struct {
+	thrift thrift.TServer
+	http   *endpoints.TwitterServer
+}
+
+func makeServers(thrift thrift.TServer, http *endpoints.TwitterServer) servers {
+	return servers{thrift, http}
+}
+
+// Creates the default MagicBag and JsonSchema for this Server and
+// returns them.  These can be modified before calling RunServer
+// Magic Bag does not include:
+// - thrift.TServerTransport
+// - *endpoints.TwitterServer
+// - *temp.TempDir
+// these should be added by callers before invoking RunServer
+func Defaults() (*ice.MagicBag, jsonconfig.Schema) {
+	bag := ice.NewMagicBag()
+	bag.PutMany(
+		thrift.NewTTransportFactory,
+		func() thrift.TProtocolFactory {
+			return thrift.NewTBinaryProtocolFactoryDefault()
+		},
+
+		endpoints.MakeStatsReceiver,
+		func() endpoints.StatScope { return "workerserver" },
+
+		// Create Execer Func
+		func() execer.Execer {
+			return execers.MakeSimExecerInterceptor(execers.NewSimExecer(nil), osexec.NewExecer())
+		},
+
+		// Create Runner Func
+		func(ex execer.Execer, tmpDir *temp.TempDir) runner.Runner {
+			outputCreator, err := localrunner.NewOutputCreator(tmpDir)
+			if err != nil {
+				log.Fatal("Error creating OutputCreatorr: ", err)
+			}
+			return localrunner.NewSimpleRunner(ex, snapshots.MakeTempCheckouter(tmpDir), outputCreator)
+		},
+		NewHandler,
+		MakeServer,
+		makeServers,
+	)
+
+	schema := jsonconfig.Schema(make(map[string]jsonconfig.Implementations))
+	return bag, schema
+}
+
+// Starts the Server based on the MagicBag and config schema provided
+// this method blocks until the server completes running or an
+// exception occurs.
+func RunServer(
+	bag *ice.MagicBag,
+	schema jsonconfig.Schema,
```

Comment:
yup, added, I totally missed that, because config isn't being used yet, but once it is we don't want to have to change this again